From 54d8978c9d05ca0416e3912ad0541a5eb52419a1 Mon Sep 17 00:00:00 2001
From: Alice Mikhaylenko <alicem@mailbox.org>
Date: Wed, 13 May 2026 22:13:53 +0400
Subject: [PATCH] libretro: Fix savestates (#1146)

Otherwise it saves all zeroes, and when loading starts reading right after
the provided data too.

Tested and working with base libretro port.
---
 src/os/libretro/StellaLIBRETRO.cxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/os/libretro/StellaLIBRETRO.cxx b/src/os/libretro/StellaLIBRETRO.cxx
index 04e914dc9..4889b6f48 100644
--- a/src/os/libretro/StellaLIBRETRO.cxx
+++ b/src/os/libretro/StellaLIBRETRO.cxx
@@ -189,6 +189,7 @@ bool StellaLIBRETRO::loadState(const void* data, size_t size)
   Serializer state;
 
   state.putByteArray(std::span{reinterpret_cast<const uInt8*>(data), size});
+  state.rewind();
 
   if(!myOSystem->state().loadState(state))
     return false;
@@ -208,6 +209,7 @@ bool StellaLIBRETRO::saveState(void* data, size_t size) const
   if (state.size() > size)
     return false;
 
+  state.rewind();
   state.getByteArray(std::span{reinterpret_cast<uInt8*>(data), state.size()});
   return true;
 }
-- 
2.53.0

